app/utils/images.js

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+let previewComponents;
+const IMAGE_MAX_HEIGHT = 350;
+const IMAGE_MIN_DIMENSION = 50;
+
+export const calculateDimensions = (height, width, viewPortWidth = 0, viewPortHeight = 0) => {
```

Comment:
Would be nice to have unit test for `calculateDimensions`

Reply:
I’ll add it